Modules/_asynciomodule.c

```diff
@@ -2089,8 +2093,14 @@ _asyncio_Task___init___impl(TaskObj *self, PyObject *coro, PyObject *loop,
         return -1;
     }
 
-    if (task_call_step_soon(state, self, NULL)) {
-        return -1;
+    if (coro_result == NULL) {
+        if (task_call_step_soon(state, self, NULL)) {
+            return -1;
+        }
+    }
+    else {
+        Py_INCREF(coro_result);
+        task_step2_impl(state, self, coro_result);
```

Comment:
The return value should be checked here, I assume if it's NULL we'll want to error out.  But there should also presumably be a test case added which will hit that.

Reply:
I'd like to find the test case that would trip on this first :)